patroni/postgresql/__init__.py

```diff
@@ -984,16 +988,30 @@ def pick_synchronous_standby(self, cluster, sync_node_count=1):
         # receiving changes faster than the sync member (very rare but possible). Such cases would
         # trigger sync standby member swapping frequently and the sort on sync_state desc should
         # help in keeping the query result consistent.
-        for app_name, state, sync_state in self.query(
-                "SELECT pg_catalog.lower(application_name), state, sync_state"
+        for app_name, state, sync_state, replica_lag in self.query(
+                "SELECT app_name, state, sync_state,"
+                "       (case when replica_count > 0 then max_replica_lsn - replica_lsn"
+                "             else current_lsn - replica_lsn"
+                "        end) replica_lag"
```

Comment:
There is no guaranty that the most up-to-date replica is actually the member of the cluster (it could be pg_receivewal for example).

Taking into account that there are also potential issues with pg_current_wal_lsn() (see below), I think it would make sense to perform post-processing in the python code.

Instead of calling pg_current_wal_lsn() you can use the values returned by the `last_operation()` method:
```python
current_wal_lsn = int(str(self.last_operation()))
```
This value could be a bit not up-to-date, but that shouldn't cause a big issue.